modules/pam_faillock/faillock.c

```diff
@@ -76,13 +76,23 @@ open_tally (const char *dir, const char *user, uid_t uid, int create)
 		flags |= O_CREAT;
 	}
 
-	fd = open(path, flags, 0600);
+	fd = open(path, flags, 0660);
 
 	free(path);
 
 	if (fd != -1) {
 		struct stat st;
 
+		/*
+		 * If umask is set to 022, as will probably in most systems, then the
+		 * group will not be able to write to the file. So, change the file
+		 * permissions just in case.
+		 * Note: owners of this file are user:root, so if the permissions are
+		 * not changed the root process writing to this file will require
+		 * CAP_DAC_OVERRIDE.
+		 */
+		ignore_return(fchmod(fd, 0660));
+
 		while (flock(fd, LOCK_EX) == -1 && errno == EINTR);
 		if (fstat(fd, &st) == 0) {
 			if (st.st_uid != uid) {
```

Comment:
Should this fchmod be made along with fchown below?

Reply:
Yes, perhaps it could also be checked whether the group write permission isn't set already as we have the information readily available.

Reply:
I have changed it based on your comments.